packages/util/test/bytes.spec.ts

```diff
@@ -351,3 +352,38 @@ tape('intToHex', function (st) {
   st.ok(intToHex(1) == '0x1', 'correctly converts 1 to a hex string')
   st.end()
 })
+
+tape('validateNoLeadingZeroes', function (st) {
+  const noLeadingZeroes = {
+    a: toBuffer('0x123'),
+    b: undefined,
+    c: toBuffer('0x'),
+  }
+  const leadingZeroes = {
+    a: toBuffer('0x001'),
+    b: toBuffer('0x'),
```

Comment:
The `b` case is wrong here I guess?

Reply:
I don't think so.  the `toBuffer` method [strips the 0x from hex strings](https://github.com/ethereumjs/ethereumjs-monorepo/blob/fbd2a320dbeed872420272e688b871fd359e1498/packages/util/src/bytes.ts#L167)) which leaves a null buffer.  Just to verify, I commented out the `a` value in the test data and wrote the value of `b` to the console and it produced 

```
b <Buffer >
not ok 86 should throw
```

I believe this is correct since the `toBufer` considers '0x' to be an empty byte array.

Reply:
Ah, had a closer look, I guess I have a problem with your test setup here.

So it's just unfortunate to throw this all in one dict (and - to add some confusion - the `leadingZeros` -> `b` naming is also not correct). With this setup it's not possible to test the single values respectively see which one goes wrong. Can you please improve here and do separate dicts for each value?